packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISequencedDocumentMessage, MessageType } from "@fluidframework/protocol-definitions";
+import { ContainerMessageType, ContainerRuntimeMessage } from "../containerRuntime";
+import { OpDecompressor } from "./opDecompressor";
+import { OpSplitter } from "./opSplitter";
+
+export class RemoteMessageProcessor {
+    constructor(
+        private readonly opSplitter: OpSplitter,
+        private readonly opDecompressor: OpDecompressor,
+    ) { }
+
+    public get partialMessages(): ReadonlyMap<string, string[]> {
+        return this.opSplitter.chunks;
+    }
+
+    public clearPartialMessagesFor(clientId: string) {
+        this.opSplitter.clearPartialChunks(clientId);
+    }
+
+    public process(remoteMessage: ISequencedDocumentMessage): ISequencedDocumentMessage {
+        let message = copy(remoteMessage);
```

Comment:
@andre4i, @vladsud, this split appears to have introduced a second shallow copy of remoteMessage.
containerRuntime.ts had a copy with a large message that is now found in `copy` defined below. containterRuntime.ts retained the copy (line 1687 in this PR) with a smaller message. The object isn't that big and it is shallow copy so defense in depth? Or accidental duplication?
@markfields was also unsure that the large comment with mention of multiple containers receiving same made sense.

Reply:
@jason-ha this was intentional. See https://github.com/microsoft/FluidFramework/pull/13273#discussion_r1042478172

Reply:
So, defense in depth then. Okay.
What about the large comment applicability? Still correct? Maybe legacy or future proofing?